File: wear/src/main/java/technology/mainthread/apps/gatekeeper/GatekeeperApp.java
package technology.mainthread.apps.gatekeeper;

import dagger.android.AndroidInjector;
import dagger.android.support.DaggerApplication;
import technology.mainthread.apps.gatekeeper.injector.component.DaggerAppComponent;
import technology.mainthread.apps.gatekeeper.injector.module.AppModule;
import technology.mainthread.apps.gatekeeper.injector.module.WearModule;
import timber.log.Timber;

public class GatekeeperApp extends DaggerApplication {

    @Override
    public void onCreate() {
        super.onCreate();

        Timber.plant(BuildConfig.DEBUG ? new Timber.DebugTree() : new ReleaseTree());
    }

    @Override
    protected AndroidInjector<? extends DaggerApplication> applicationInjector() {
        return DaggerAppComponent.builder()
                .appModule(new AppModule(this))
                .wearModule(new WearModule(this))
                .create(this);
    }
}

File: wear/src/main/java/technology/mainthread/apps/gatekeeper/GatekeeperApp.kt
package technology.mainthread.apps.gatekeeper

import dagger.android.AndroidInjector
import dagger.android.support.DaggerApplication
import technology.mainthread.apps.gatekeeper.injector.component.DaggerAppComponent
import technology.mainthread.apps.gatekeeper.injector.module.AppModule
import technology.mainthread.apps.gatekeeper.injector.module.WearModule
import timber.log.Timber

class GatekeeperApp : DaggerApplication() {

    override fun onCreate() {
        super.onCreate()

        Timber.plant(if (BuildConfig.DEBUG) Timber.DebugTree() else ReleaseTree())
    }

    override fun applicationInjector(): AndroidInjector<out DaggerApplication> {
        return DaggerAppComponent.builder()
                .appModule(AppModule(this))
                .wearModule(WearModule(this))
                .create(this)
    }
}
